pkg/oci/oci.go

```diff
@@ -73,15 +77,33 @@ func getLocalOciStore() (*oci.Store, error) {
 	return oci.New(defaultOciStore)
 }
 
-// GetGadgetImage pulls the gadget image and returns the a structure representing it.
-func GetGadgetImage(ctx context.Context, image string, authOpts *AuthOptions) (*GadgetImage, error) {
+// GetGadgetImage pulls the gadget image according to the pull policy and returns
+// a GadgetImage structure representing it.
+func GetGadgetImage(ctx context.Context, image string, authOpts *AuthOptions, pullPolicy string) (*GadgetImage, error) {
```

Comment:
I was thinking if we should do the same for `GetEbpfObject` and `GetMetadata`, but they aren't used anymore. Perhaps you can remove them in this PR too?